fix: prevent None entrance_exam_minimum_score_pct from breaking CourseOverview sync

Abdul-Muqadim-Arbisoft opened the pull request:
When entrance exams are disabled in Studio, the field `entrance_exam_minimum_score_pct` was set to `None`. This caused silent failures when saving `CourseOverview` because the database column requires a float (NOT NULL).

This patch ensures that:
- CourseOverview sanitizes None values by falling back to `settings.ENTRANCE_EXAM_MIN_SCORE_PCT` (default=50).
- Studio avoids writing `None` and instead applies the configured default.

Impact:
- Prevents IntegrityErrors and silent failures when updating course settings.
- Restores proper syncing between modulestore (Mongo) and CourseOverview (MySQL).
- Fixes reported issues such as display name changes not persisting and course start dates not syncing.

Closes: https://github.com/openedx/edx-platform/issues/37319#

Abdul-Muqadim-Arbisoft committed:
fix: prevent None entrance_exam_minimum_score_pct from breaking CourseOverview sync

When entrance exams are disabled in Studio, the field
`entrance_exam_minimum_score_pct` was set to `None`. This caused silent failures
when saving `CourseOverview` because the database column requires a float (NOT NULL).

This patch ensures that:
- CourseOverview sanitizes None values by falling back to
  `settings.ENTRANCE_EXAM_MIN_SCORE_PCT` (default=50).
- Studio avoids writing `None` and instead applies the configured default.

Impact:
- Prevents IntegrityErrors and silent failures when updating course settings.
- Restores proper syncing between modulestore (Mongo) and CourseOverview (MySQL).
- Fixes reported issues such as display name changes not persisting and course
  start dates not syncing.

Closes: https://github.com/openedx/edx-platform/issues/37319#

openedx-webhooks commented:
Thanks for the pull request, @Abdul-Muqadim-Arbisoft!


This repository is currently maintained by `@openedx/wg-maintenance-edx-platform`.


Once you've gone through the following steps feel free to tag them in a comment and let them know that your changes are ready for engineering review.

<details><summary>:radio_button: Get product approval</summary>

If you haven't already, [check this list](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Does-my-contribution-require-Product-Review%3F) to see if your contribution needs to go through the product review process.

- If it does, you'll need to submit a product proposal for your contribution, and have it reviewed by the [Product Working Group](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3449028609/Product+Working+Group).
    - This process (including the steps you'll need to take) is documented [here](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Product-Review-Process).
- If it doesn't, simply proceed with the next step.
</details>

<details><summary>:radio_button: Provide context</summary>

To help your reviewers and other members of the community understand the purpose and larger context of your changes, feel free to add as much of the following information to the PR description as you can:

- Dependencies
  > This PR must be merged before / after / at the same time as ...
- Blockers
  > This PR is waiting for OEP-1234 to be accepted.
- Timeline information
  > This PR must be merged by XX date **because** ...
- Partner information
  > This is for a course on edx.org.
- Supporting documentation
- Relevant [Open edX discussion forum](https://discuss.openedx.org/) threads
</details>


<details><summary>:radio_button: Submit a signed contributor agreement (CLA)</summary>

:warning: We ask all contributors to the Open edX project to submit a [signed contributor agreement](https://openedx.org/cla) or indicate their institutional affiliation.
Please see the [CONTRIBUTING](https://github.com/openedx/.github/blob/master/CONTRIBUTING.md) file for more information.

If you've signed an agreement in the past, you may need to re-sign.
See [The New Home of the Open edX Codebase](https://open.edx.org/blog/the-new-home-of-the-open-edx-codebase/) for details.

Once you've signed the CLA, please allow 1 business day for it to be processed.
After this time, you can re-run the CLA check by adding a comment below that you have signed it.
If the CLA check continues to fail, you can tag the `@openedx/cla-problems` team in a comment for further assistance.
</details>
<!-- comment:no_cla -->




<details><summary>:radio_button: Get a green build</summary>

If one or more checks are failing, continue working on your changes until this is no longer the case and your build turns green.
</details><details>


---

<details><summary>Where can I find more information?</summary>

If you'd like to get more details on all aspects of the review process for open source pull requests (OSPRs), check out the following resources:

- [Overview of Review Process for Community Contributions](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/FAQ-about-pull-requests.html)
- [Pull Request Status Guide](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/pull-request-statuses.html)
- [Making changes to your pull request](https://docs.openedx.org/en/latest/documentors/how-tos/make_changes_to_your_pull_request.html)
</details>

<details><summary>When can I expect my changes to be merged?</summary>

Our goal is to get community contributions seen and reviewed as efficiently as possible.

However, the amount of time that it takes to review and merge a PR can vary significantly based on factors such as:

- The size and impact of the changes that it introduces
- The need for product review
- Maintenance status of the parent repository

:bulb: *As a result it may take up to several weeks or months to complete a review and merge your PR.*
</details>
<!-- comment:external_pr -->
<!-- data: eyJkcmFmdCI6IGZhbHNlfQ== -->

taimoor-ahmed-1 commented on openedx/core/djangoapps/content/course_overviews/models.py:
```diff
@@ -272,6 +272,15 @@ def _create_or_update(cls, course):  # lint-amnesty, pylint: disable=too-many-st
+        elif isinstance(course.entrance_exam_minimum_score_pct, int):
+            course_overview.entrance_exam_minimum_score_pct = course.entrance_exam_minimum_score_pct / 100
+        else:
+            course_overview.entrance_exam_minimum_score_pct = course.entrance_exam_minimum_score_pct
```
although I am not sure why `int` is checked twice, but this code block can be simplified to this:
```
if course.entrance_exam_minimum_score_pct is None:
    ee_min_score_pct = float(settings.ENTRANCE_EXAM_MIN_SCORE_PCT)
else:
     ee_min_score_pct = course.entrance_exam_minimum_score_pct

if isinstance(ee_min_score_pct, int) or (isinstance(ee_min_score_pct, float) and ee_min_score_pct.is_integer()):
     ee_min_score_pct =  ee_min_score_pct / 100

course_overview.entrance_exam_minimum_score_pct =  ee_min_score_pct
```
Please also include an inline comment on why this conditional block is being added.

Abdul-Muqadim-Arbisoft replied:
@taimoor-ahmed-1 code block simplified and inline comment added

taimoor-ahmed-1 replied:
have you tested the changes?

Abdul-Muqadim-Arbisoft replied:
Tested

mphilbrick211 commented:
Hi @Abdul-Muqadim-Arbisoft! If you have not done so already, please have your manager reach out to oscm@axim.org to have you added to our existing entity agreement. This will allow the CLA check to turn green. Thanks!

Abdul-Muqadim-Arbisoft committed:
refactor: clean up entrance_exam_minimum_score_pct handling

- Consolidate logic to avoid repeated assignments
- Centralize None fallback and int/float normalization
- Improve readability with inline comment and consistency with Open edX style

Abdul-Muqadim-Arbisoft committed:
test: update entrance exam deletion test to expect default min score

- Adjusted `test_entrance_exam_created_updated_and_deleted_successfully` to check for
  `settings.ENTRANCE_EXAM_MIN_SCORE_PCT` instead of `None` after exam deletion
- Added handling for both int and float defaults (`/100` for integer case)

taimoor-ahmed-1 reviewed (approved):
LGTM

mphilbrick211 commented:
@taimoor-ahmed-1 @Abdul-Muqadim-Arbisoft if you re-run the checks, the CLA check should turn green.
